### PR TITLE
[feat] add optional versioning id for s3 archive

### PR DESCRIPTION
**Pull request by Ryang20718**
This PR adds file versioning for `s3_archive` in order to solve the problem defined below

Consider the following problem:

if person A updates the object in the s3 bucket, it will break person's B code since s3 archive now points at the new object. with this versioning enabled, we could update the object, but not break person B's code

The `file_version` field can be added to s3_archive if the s3 bucket is versioned

```
    s3_archive(
		<kwargs>
        file_version = "SI56h6qfu3O6AYdy5soRntQvajXqRXbe3",
    )
```
resolves https://github.com/1e100/cloud_archive/issues/3

Testing:

- [x] Regression testing s3 archives without file version 
- [x] Regression testing s3 archives with bad sha256 
- [x] Regression testing s3 archives with file version 
- [x] Regression testing s3 archives with file version bad 256 
- [x] Regression testing s3 archives with file version that doesn't exist 

Sample error message if version id is invalid
```
An error occurred (InvalidArgument) when calling the GetObject operation: Invalid version id specified
```

### CODE REVIEW

File: cloud_archive.bzl [hunk 1]
```diff
@@ -95,8 +96,11 @@ def cloud_archive_download(
     elif provider == "s3":
         tool_path = repo_ctx.which("aws")
         extra_flags = ["--profile", profile] if profile else []
-        src_url = "s3://{}/{}".format(bucket, file_path)
-        cmd = [tool_path] + extra_flags + ["s3", "cp", src_url, "."]
+        bucket_arg = ["--bucket", bucket]
+        file_arg = ["--key", file_path]
+        file_version_arg = ["--version-id", file_version] if file_version else []
+        src_url = filename
+        cmd = [tool_path] + extra_flags + ["s3api", "get-object"] + bucket_arg + file_arg + file_version_arg + [filename]
```
Comment (Ryang20718) [inline cloud_archive.bzl:103]: `aws s3` doesn't support versioning. That's why I had to shift to `s3api`

**Comment by Ryang20718** [comment]
@1e100 when you have a moment, would appreciate a review :)

**Review by 1e100 — APPROVED** [review]
Looks straightforward enough, LGTM

**Comment by Ryang20718** [comment]
thanks! when you have a moment, if you could cut a new release, would be much appreciated!

**Pull request merged**